core/src/main/scala/akka/kafka/internal/ConsumerStage.scala

```diff
@@ -190,7 +192,8 @@ private[kafka] object ConsumerStage {
       if (offsets.isEmpty)
         Future.successful(Done)
       else {
-        stages.head._2.commit(this)
+        implicit val ec = akka.dispatch.ExecutionContexts.global()
+        Future.sequence(stages.map(_._2.commit(this))).map(_ => Done)
```

Comment:
Perhaps this could still do `stages.head._2.commit(this)` and then the implementation of the `Committer` can delegate to the rest of the stages, since there I think we have `materializer.executionContext` or similar.